core/domain/exp_domain.py

```diff
@@ -2460,6 +2513,48 @@ def get_all_html_content_strings(self):
 
         return html_list
 
+    @classmethod
+    def to_proto(
+            cls, exploration_id, title, version, init_state_name, states):
+        """Calculate the exploration size by setting exploration proto.
+
+        Args:
+            exploration_id: str. The id of the exploration.
+            title: str. The exploration title.
+            version: int. The version of the exploration.
+            init_state_name: str|None. The name of the initial state.
+            states: list|None. The list of state
+                domain objects.
+
+        Returns:
+            Proto Object. The exploration proto object.
+        """
+        state_protos = {}
+        exploration_proto = {}
+        if states is not None:
+            state_protos = state_domain.State.to_proto(states)
```

Comment:
This construction is a bit weird. Based on the naming, State.to_proto() should only convert one state. If typically you need to convert all states at once, then that is a function that should be in the Exploration object (I assume we don't have a States object.)

Also, when is states none?

Also shouldn't this be using self.title, self.exploration_id, self.states, etc. (just like to_dict)?

Reply:
Done, this looks much cleaner now.